backend/src/services/model.ts

```diff
@@ -32,10 +32,21 @@ export async function createModel(user: UserInterface, modelParams: CreateModelP
   })
 
   const auth = await authorisation.model(user, model, ModelAction.Create)
+
   if (!auth.success) {
     throw Forbidden(auth.info, { userDn: user.dn })
   }
 
+  if (
+    (model.settings.mirror.sourceModelId && model.settings.mirror.destinationModelId) ||
+    (modelParams.settings &&
```

Comment:
This is equivalent to `modelParams?.settings?.mirror?.destinationModelId && modelParams?.settings?.mirror?.sourceModelId`.

Also, I think checking _both_ sets here are unnecessarily because the `modelParams` are directly used to fill in the `model` value.

Also, we should split this out into a separate function and use it on both `create` and `update` events to reduce duplication.